automation/taskcluster/lib/artifacts.py

```diff
@@ -0,0 +1,22 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import re
+import subprocess
+
+
+def from_gradle():
+    process = subprocess.Popen(["./gradlew", "--no-daemon", "printModules"], stdout=subprocess.PIPE)
+    (output, err) = process.communicate()
+    exit_code = process.wait()
+
+    if exit_code is not 0:
+        print "Gradle command returned error:", exit_code
+
+    tuples = re.findall('module: name=(.*) buildPath=(.*)', output, re.M)
+    return map(lambda (name, build_path): {
+        'name': name[1:],
```

Comment:
I'll add a comment, great point!

Unrelated, but I noticed that `decision_task` and `decision_task_release` both handle the name a little differently: `decision_task` loves to keep the `:`, but `decision_task_release` uses name without the `:` in a lot of places. I didn't investigate fully, but since, in the end, it's just creating a graph for `taskcluster`, I'd love to make these more consistent. Probably not something for this PR, but a bit of a :thinking:

Reply:
Good catch on making these consistent! +1 to that (in a followup PR). For the record, we haven't investigated too much into making these graphs generic across Android projects. That's something we'd like to invest on.